cpp/tests/binning/bin_tests.cpp

```diff
@@ -0,0 +1,429 @@
+
+TYPED_TEST(NegativeNumbersBinTestFixture, TestNegativeNumbers256) { this->test(256); };
+TYPED_TEST(NegativeNumbersBinTestFixture, TestNegativeNumbers512) { this->test(512); };
+TYPED_TEST(NegativeNumbersBinTestFixture, TestNegativeNumbers1024) { this->test(1024); };
+
+/*
+ * Test fixed point types.
+ */
+
+template <typename T>
+struct FixedPointBinTestFixture : public BinTestFixture {
+};
+
+TYPED_TEST_CASE(FixedPointBinTestFixture, FixedPointTypes);
+
+TYPED_TEST(FixedPointBinTestFixture, TestFixedPointData)
+{
+  using fpc_type_wrapper = fpc_wrapper<cudf::device_storage_type_t<TypeParam>>;
+
+  fpc_type_wrapper left_edges{{0, 10, 20, 30, 40, 50, 60, 70, 80, 90}, numeric::scale_type{0}};
+  fpc_type_wrapper right_edges{{10, 20, 30, 40, 50, 60, 70, 80, 90, 100}, numeric::scale_type{0}};
+  fpc_type_wrapper input{{25, 25, 25, 25, 25, 25, 25, 25, 25, 25}, numeric::scale_type{0}};
+
+  auto result =
+    cudf::bin(input, left_edges, cudf::inclusive::YES, right_edges, cudf::inclusive::NO);
+
+  // Check that every element is placed in bin 2.
+  fwc_wrapper<cudf::size_type> expected{{2, 2, 2, 2, 2, 2, 2, 2, 2, 2},
+                                        {1, 1, 1, 1, 1, 1, 1, 1, 1, 1}};
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->view());
+};
+
+/*
+ * Test strings.
+ */
+
+// Basic test of strings of lowercase alphanumerics.
+TEST(TestStringData, SimpleStringTest)
+{
+  strings_column_wrapper left_edges{"a", "b", "c", "d", "e"};
+  strings_column_wrapper right_edges{"b", "c", "d", "e", "f"};
+  strings_column_wrapper input{"abc", "bcd", "cde", "def", "efg"};
+
+  auto result =
+    cudf::bin(input, left_edges, cudf::inclusive::YES, right_edges, cudf::inclusive::NO);
+
+  fwc_wrapper<cudf::size_type> expected{{0, 1, 2, 3, 4}, {1, 1, 1, 1, 1}};
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->view());
+};
+
+// Test non-ASCII characters.
+TEST(TestStringData, NonAsciiStringTest)
+{
+  strings_column_wrapper left_edges{"A"};
+  strings_column_wrapper right_edges{"z"};
+  strings_column_wrapper input{"Héllo",
+                               "thesé",
+                               "HERE",
+                               "tést strings",
+                               "",
+                               "1.75",
+                               "-34",
+                               "+9.8",
+                               "17¼",
+                               "x³",
+                               "2³",
+                               " 12⅝",
+                               "1234567890",
+                               "de",
+                               "\t\r\n\f "};
+
+  auto result = cudf::bin(input, left_edges, cudf::inclusive::NO, right_edges, cudf::inclusive::NO);
+
+  fwc_wrapper<cudf::size_type> expected{{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+                                        {1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0}};
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->view());
+}
+
+}  // anonymous namespace
+
```

Comment:
The only other gtests I would recommend is to add some sliced column views. The current code should handle this fine because of the begin/end iterators automatically account for sliced views. It would be good to have 1 or 2 in case the code is changed in the future.